### PR TITLE
Drop nose in favor of py.test

### PR DESCRIPTION
**Pull request by elyezer**
py.test is now a Robottelo dependency. Because that we can now drop nose
and start using py.test to run tests and generate the junit report.

### CODE REVIEW

**Comment by omaciel** [comment]
ACK

**Comment by sthirugn** [comment]
@omaciel can you confirm that the behavior intended by https://github.com/SatelliteQE/robottelo/pull/3089/files is still intact with this change?

**Comment by sthirugn** [comment]
ACK pending question

**Comment by omaciel** [comment]
It does look like we need to add `-s` so not to capture the stdout.

**Comment by omaciel** [comment]
@elyezer ping?

**Comment by omaciel** [comment]
@elyezer ping x 2?

**Comment by elyezer** [comment]
@omaciel I've run the tests in tmux and the buffer was not enough to get to the beginning of the test run... Running again.

**Comment by elyezer** [comment]
I've run a small set of API tests.

```
$ py.test tests/foreman/api/test_{architecture,config_template}.py                                          
================================================================= test session starts =================================================================
platform linux2 -- Python 2.7.5, pytest-2.8.3, py-1.4.31, pluggy-0.3.1
rootdir: /home/elyezer/robottelo, inifile: 
plugins: cov-2.2.0, xdist-1.13.1
collected 14 items 

tests/foreman/api/test_architecture.py ......s
tests/foreman/api/test_config_template.py .......

======================================================== 13 passed, 1 skipped in 33.72 seconds ========================================================
```

Below is the output running in threaded mode

```
$ py.test -n auto --boxed tests/foreman/api/test_{architecture,config_template}.py
================================================================= test session starts =================================================================
platform linux2 -- Python 2.7.5, pytest-2.8.3, py-1.4.31, pluggy-0.3.1
rootdir: /home/elyezer/robottelo, inifile: 
plugins: cov-2.2.0, xdist-1.13.1
gw0 [14] / gw1 [14] / gw2 [14] / gw3 [14] / gw4 [14] / gw5 [14] / gw6 [14] / gw7 [14]
scheduling tests via LoadScheduling
s.............
======================================================== 13 passed, 1 skipped in 19.01 seconds ========================================================
```

And here is in verbose mode which I think is better than printing all logging messages on the stdout

```
$ py.test -v tests/foreman/api/test_{architecture,config_template}.py
================================================================= test session starts =================================================================
platform linux2 -- Python 2.7.5, pytest-2.8.3, py-1.4.31, pluggy-0.3.1 -- /home/elyezer/.virtualenvs/robottelo/bin/python
cachedir: .cache
rootdir: /home/elyezer/robottelo, inifile: 
plugins: cov-2.2.0, xdist-1.13.1
collected 14 items 

tests/foreman/api/test_architecture.py::ArchitectureTestCase::test_associate_with_os PASSED
tests/foreman/api/test_architecture.py::ArchitectureTestCase::test_negative_create_name PASSED
tests/foreman/api/test_architecture.py::ArchitectureTestCase::test_negative_update_name PASSED
tests/foreman/api/test_architecture.py::ArchitectureTestCase::test_positive_create_name PASSED
tests/foreman/api/test_architecture.py::ArchitectureTestCase::test_positive_delete PASSED
tests/foreman/api/test_architecture.py::ArchitectureTestCase::test_positive_update_name PASSED
tests/foreman/api/test_architecture.py::ArchitectureTestCase::test_post_hash <- robottelo/decorators.py SKIPPED
tests/foreman/api/test_config_template.py::ConfigTemplateTestCase::test_build_pxe_default <- robottelo/decorators.py PASSED
tests/foreman/api/test_config_template.py::ConfigTemplateTestCase::test_config_template_orgs PASSED
tests/foreman/api/test_config_template.py::ConfigTemplateTestCase::test_negative_create_name PASSED
tests/foreman/api/test_config_template.py::ConfigTemplateTestCase::test_negative_update_name PASSED
tests/foreman/api/test_config_template.py::ConfigTemplateTestCase::test_positive_create_name PASSED
tests/foreman/api/test_config_template.py::ConfigTemplateTestCase::test_positive_delete PASSED
tests/foreman/api/test_config_template.py::ConfigTemplateTestCase::test_positive_update_name PASSED

======================================================== 13 passed, 1 skipped in 33.14 seconds ========================================================
```

Should we run in verbose mode?

**Comment by omaciel** [comment]
+1 for running with verbose mode

**Comment by elyezer** [comment]
Added the `-v` to the `PYTEST_OPTS`

**Comment by omaciel** [comment]
ACK again

**Comment by oshtaier** [comment]
Does it mean that it will not be possible to run tests using nosetests anymore?

**Comment by omaciel** [comment]
@oshtaier the change only affects running tests via the Make file

File: Makefile [hunk 1]
```diff
@@ -8,9 +8,9 @@ FOREMAN_RHCI_TESTS_PATH=$(join $(FOREMAN_TESTS_PATH), rhci)
 FOREMAN_SMOKE_TESTS_PATH=$(join $(FOREMAN_TESTS_PATH), smoke)
 FOREMAN_TESTS_PATH=tests/foreman/
 FOREMAN_UI_TESTS_PATH=$(join $(FOREMAN_TESTS_PATH), ui)
-NOSETESTS=python -m cProfile -o $@.pstats $$(which nosetests)
-NOSETESTS_OPTS=-s --logging-filter=nailgun,robottelo --with-xunit\
-			   --xunit-file=foreman-results.xml
+PYTEST=python -m cProfile -o $@.pstats $$(which py.test)
+PYTEST_OPTS=-v --junit-xml=foreman-results.xml
```
Comment (kbidarkar) [inline Makefile]: :+1:  for running with -v , we can quickly know if some test passed , we do not have to wait for all the tests to complete.

Also anything which shows progress of the automation, is good, not necessarily detailed logging is required.
We can anyways refer individual logs for it. :)

**Comment by sthirugn** [comment]
@kbidarkar acked in irc. ACK again from me.

@elyezer rebase?

**Pull request merged**